src/types/DataTypes.sol

```diff
@@ -241,42 +255,28 @@ library LockupLinear {
 
     /// @notice Struct encapsulating the time range.
     /// @param start The Unix timestamp for the stream's start.
-    /// @param cliff The Unix timestamp for the cliff period's end.
+    /// @param cliff The Unix timestamp for the cliff period's end. A value of zero means there is no cliff.
     /// @param end The Unix timestamp for the stream's end.
     struct Range {
```

Comment:
> This will further move getCliffTime() .. into the common logic. All dynamic streams will have 0 cliff.

I don't think it's a good idea to populate all child contracts with the `_cliffs` mapping storage property. This is going to be ignored in all contracts except for Linear.

> Plus if we include (https://github.com/sablier-labs/v2-core/pull/813#discussion_r1486373028) both Range and Segments into Stream struct inside LockupDynamic and LockupLinear namespaces, we can further move getStream() into common logic

No, this is not correct. It is impossible to include `getStream` into the common logic because each child contract of `SablierV2Lockup` has its unique storage properties.

I think you are missing the fact that [LockupLinear.Range](https://github.com/sablier-labs/v2-core/blob/394eb5fee10a4e21b6e5c50d21fa2cb24d05aa52/src/types/DataTypes.sol#L260-L264) and [LockupDynamic.Range](https://github.com/sablier-labs/v2-core/blob/394eb5fee10a4e21b6e5c50d21fa2cb24d05aa52/src/types/DataTypes.sol#L155-L158) are different. It doesn't make sense to include segments in the Linear namespace.

Please see again the design I've proposed [here](https://github.com/sablier-labs/v2-core/discussions/805). If you considered it bad, comment on that discussion.

Reply:
> I don't think it's a good idea to populate all child contracts with the _cliffs mapping storage property. This is going to be ignored in all contracts except for Linear.

I agree with this. This was just an idea I had while reviewing the PR but I will discard it now as it looks like an overkill.

> It is impossible to include getStream into the common logic because each child contract of SablierV2Lockup has its unique storage properties.

Agree, it's not possible because `_cliffs` and `segments` are unique to Linear and Dynamic respectively. 

> I think you are missing the fact that [LockupLinear.Range](https://github.com/sablier-labs/v2-core/blob/394eb5fee10a4e21b6e5c50d21fa2cb24d05aa52/src/types/DataTypes.sol#L260-L264) and [LockupDynamic.Range](https://github.com/sablier-labs/v2-core/blob/394eb5fee10a4e21b6e5c50d21fa2cb24d05aa52/src/types/DataTypes.sol#L155-L158) are different.

I am aware of that.

> If you considered it bad

No, I didn't consider it bad but quite the opposite.